backend-processing.ts

```diff
@@ -96,24 +99,65 @@ export default class BackendProcessing {
     }
     let url = entry["hyperlink url"];
     let comments = entry["comments"];
-    database.insertFromCsvLine(
-      userID,
-      fundID,
-      firstCompany,
-      name,
-      firstPosition,
-      sterm,
-      eterm,
-      employer,
-      position,
-      url,
-      comments
-    );
+    database
+      .insertFromCsvLine(userID, fundID, firstCompany, name, firstPosition, sterm, eterm, employer, position, url, comments)
+      .then((individualID) => {
+        let q1 = entry["q1 " + year];
+        let q2 = entry["q2 " + year];
+        let q3 = entry["q3 " + year];
+        let q4 = entry["q4 " + year];
+        this.processQuarter(q1, 1, individualID, userID, fundID, year);
+        this.processQuarter(q2, 2, individualID, userID, fundID, year);
+        this.processQuarter(q3, 3, individualID, userID, fundID, year);
+        this.processQuarter(q4, 4, individualID, userID, fundID, year);
+      });
+  }
+  processQuarter(quarterString, quarterNumber, individualID, userID, fundID, year) {
+    if (quarterString == null || quarterString == "") return;
+    let quarterDates = [year + "/01/01", year + "/04/01", year + "/07/01", year + "/10/01"];
+    let positionName = "";
+    let companyName = "";
+    let p = quarterString.indexOf(" at ");
+    if (p > 0) {
+      positionName = quarterString.substring(0, p);
+      companyName = quarterString.substring(p + 4);
+    }
+    p = quarterString.indexOf(",");
+    if (p > 0) {
+      positionName = quarterString.substring(0, p);
+      companyName = quarterString.substring(p + 1);
+    }
```

Comment:
I believe that these two scenarios for the positionName and companyName assignments are mutually exclusive. If not, please make any necessary corrections.

Reply:
Just communicated with my partner, it seems these are mutually exclusive, so no problem.